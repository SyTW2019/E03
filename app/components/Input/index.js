import React from "react";

export default class Input extends React.Component {
  render(){
    return(
      <div>
        <input 
          text={this.props.text}
          ref={this.props.ref}
          type={this.props.type}
          value={this.props.value}
          onChange={this.props.onChange}
          />
      </div>
    );
  }
}
