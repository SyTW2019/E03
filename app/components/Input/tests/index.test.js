import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from 'react-testing-library';
import "jest-dom/extend-expect";
import renderer from "react-test-renderer";

import Input from '../index';

const name='Prueba';
const ref='/Prueba';
const type='otraPrueba';
const value='valorPrueba';
const onChange='funcionPrueba';
const renderComponent = ( props = {} ) => render(<Input name={name} {...props}/>);

describe('<Input />', () => {

  afterEach(cleanup);

  it('render', () => {
    const div = document.createElement(div);
    ReactDOM.render(<Input/>,div);
  });

  it('should have an name attribute', () => {
    const { container } = renderComponent();
    const element = container.querySelector('input');
    expect(element.hasAttribute('name')).toBe(true);
  });

  it('Matches snapshots', () => {
    const tree = renderer.create(<Input/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
