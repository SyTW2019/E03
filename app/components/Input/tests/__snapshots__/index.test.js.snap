// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Matches snapshots 1`] = `
<div>
  <input
    data-testid="input"
  />
</div>
`;
